include/mpp/algorithm/inverse.hpp

```diff
@@ -120,57 +125,70 @@ namespace mpp
+				make_identity_buffer<false>(l_buffer,
+					rows,
 					columns,
-					l_buffer,
-					u_buffer);
+					default_floating_type{},
+					default_floating_type{ 1 });
+
+				const auto det = lu_generic<default_floating_type, true, true>(rows, columns, l_buffer, u_buffer);
 
 				if constexpr (Check)
 				{
 					if (floating_point_compare(det, default_floating_type{}) ==
 						default_floating_type_ordering_type::equivalent)
 					{
-						throw std::runtime_error(INVERSE_MATRIX_SINGULAR);
+						throw std::runtime_error(MATRIX_SINGULAR);
 					}
 				}
 
-				if (std::is_constant_evaluated())
-				{
-					forward_substitution_in_place(l_buffer, columns);
-					back_substitution_in_place(u_buffer, columns);
-				}
-				else
-				{
-					// Compute inv(L) and inv(U) in parallel because they don't share data
+				// Solve for x_buffer values with Ax=b with A=l_buffer and b=Column of identity matrix
 
-					auto l_inverse_future = std::async(std::launch::async, [&l_buffer, columns]() {
-						forward_substitution_in_place(l_buffer, columns);
-					});
-					auto u_inverse_future = std::async(std::launch::async, [&u_buffer, columns]() {
-						back_substitution_in_place(u_buffer, columns);
-					});
+				auto identity_column_buffer = typename matrix<default_floating_type, RowsExtent, 1>::buffer_type{};
 
-					l_inverse_future.wait();
-					u_inverse_future.wait();
-				}
+				allocate_buffer_if_vector(identity_column_buffer, rows, 1, default_floating_type{});
 
-				matrix_multiplication_on_buffers<To, default_floating_type>(inverse_matrix_buffer,
-					std::move(u_buffer),
-					std::move(l_buffer),
-					rows,
-					columns,
-					rows,
-					columns);
+				for (auto row = std::size_t{}; row < rows; ++row)
+				{
+					// Branch-less logic to do:
+					// row if row == 0
+					// row - 1 if row > 0
+					const auto last_column_index = row - 1 * (row != 0);
+
+					// This forms b that corresponds to corresponding column from the identity matrix
+					// e.g. with a 3x1 column vector
+					// row 1 -> 1 0 0
+					// row 2 -> 0 1 0
+					// row 3 -> 0 0 1
+					identity_column_buffer[last_column_index] = default_floating_type{};
+					identity_column_buffer[row]               = default_floating_type{ 1 };
+
+					auto l_x_buffer = forward_subst_on_buffer_unchecked<default_floating_type, RowsExtent, 1>(l_buffer,
+						identity_column_buffer,
+						rows);
+
+					// Use l_x_buffer to do back substitution to solve Ax=B with A=u_buffer and b=l_x_buffer. The
+					// part_inverse_buffer now corresponds to a column of the inverse matrix
+
+					auto part_inverse_buffer =
+						back_subst_on_buffer_unchecked<default_floating_type, RowsExtent, 1>(u_buffer,
+							std::move(l_x_buffer),
+							rows);
+
+					for (auto column = std::size_t{}; auto&& value : part_inverse_buffer)
+					{
+						inv_buffer[index_2d_to_1d(columns, column++, row)] = std::move(value);
+					}
+				}
```

Comment:
**NOTE TO SELF**: This process of the for loop can be parallelized. Need to make an issue noting this.

Reply:
Noted